central/authprovider/datastore/datastore_impl.go

```diff
@@ -86,12 +86,18 @@ func (b *datastoreImpl) verifyExistsAndMutable(ctx context.Context, id string, f
 		return errox.NotFound.Newf("auth provider with id %q was not found", id)
 	}
 
-	if provider.GetTraits().GetMutabilityMode() == storage.Traits_ALLOW_MUTATE {
+	switch provider.GetTraits().GetMutabilityMode() {
+	case storage.Traits_ALLOW_MUTATE:
 		return nil
+	case storage.Traits_ALLOW_MUTATE_FORCED:
+		if force {
+			return nil
+		}
+		return errox.InvalidArgs.Newf("auth provider %q is immutable and can only be removed"+
+			" via API and specifying the force flag", id)
+	default:
+		utils.Should(errors.Wrapf(errox.InvalidArgs, "unknown mutability mode given: %q",
+			provider.GetTraits().GetMutabilityMode().String()))
```

Comment:
```suggestion
			provider.GetTraits().GetMutabilityMode()))
```
`%q` and `%s` implicitly convert to `string`, via `fmt.Stringer` or casting to `string` as underlying type, if possible